src/main/java/justin/Parser.java

```diff
@@ -0,0 +1,182 @@
+     * @return an AddToDoCommand.
+     * @throws DukeException if the description of the ToDo is empty.
+     */
+    public static Command parseToDo(String message) throws DukeException{
+        Scanner sc = new Scanner(message);
+        if (sc.hasNext()) {
+            String description = sc.nextLine();
+            return new AddToDoCommand(description, false);
+        } else {
+            throw new DukeException("OOPS! The description of the todo should not be empty!");
+        }
+    }
+
+    /**
+     * Returns an AddDeadlineCommand through scanning the description.
+     * @param message The description of the command.
+     * @return an AddDeadlineCommand.
+     * @throws DukeException if the message given is invalid.
+     */
+    public static Command parseDeadline(String message) throws DukeException {
+        Scanner sc = new Scanner(message);
+        try {
+            String description = "";
+            while (!sc.hasNext("/by")) {
+                String next = sc.next();
+                description += next + " ";
+            }
+            sc.next();
+            String next = sc.nextLine();
+            String[] date = next.split(" ");
+            return new AddDeadlineCommand(description, date);
+        } catch (Exception e) {
+            throw new DukeException("OOPS!! Please enter a valid message!");
+        }
+    }
+
+    /**
+     * Returns an AddEventCommand through scanning the description.
+     * @param message The description of the command.
+     * @return an AddEventCommand.
+     * @throws DukeException if the message is invalid.
+     */
+    public static Command parseEvent(String message) throws DukeException {
+        Scanner sc = new Scanner(message);
+        try {
+            String description = "";
+            while (!sc.hasNext("/at")) {
+                String next = sc.next();
+                description += next + " ";
+            }
+            sc.next();
+            String next = sc.nextLine();
+            String[] date = next.split(" ");
+            return new AddEventCommand(description, date);
+        } catch (Exception e) {
+            throw new DukeException("OOPS! Please enter a valid message!");
+        }
+    }
+
+
+    public static Command parseFind(String message) {
+        Scanner sc = new Scanner(message);
+        String res = "";
+        while (sc.hasNext()) {
+            res += sc.next() + " ";
+        }
+        res = res.substring(0, res.length() - 1);
+        return new FindCommand(res);
+    }
+
+    /**
+     * Returns a command from scanning the inputs of users.
+     * @param message The description of the command.
+     * @return a Command.
+     * @throws DukeException if the input is invalid.
+     */
+    public static Command parse(String message) throws DukeException {
+        Scanner sc = new Scanner(message);
+        String first = sc.next();
+        switch (first) {
+            case "bye": {
```

Comment:
For our coding standards, I think we're having the `case` clauses on the same indent level as the `switch` statements. So maybe something like this?

```java
        // ...
        switch (first) {
        case "bye": {
            return new ExitCommand();
        }
        // ...
```

If you are using an IDE, I think it should be configurable somewhere in the settings.

Reply:
Second this, can right click and select "column selection mode" in Intellij to position case clause at the same indent level as switch statements.